### PR TITLE
Implement `ConnectivityService` for mock interface

### PR DESCRIPTION
**Pull request by altonen**
This PR syncs the mock interface with latest changes and implements the `ConnectivityService`. The swarm tests are rewritten to test both libp2p and the mock interface for better test coverage.

Syncing, pubsub and peer banning in the backend are implemented in separate PRs

### CODE REVIEW

File: p2p/src/net/mock/socket.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+// Author(s): A. Altonen
+use crate::net::mock::types::Message;
+use bytes::{Buf, BytesMut};
+use serialization::{Decode, Encode};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+use tokio_util::codec::{Decoder, Encoder};
+use utils::ensure;
+
+const MAX: usize = 10 * 1024 * 1024;
```
Comment (TheQuantumPhysicist) [inline p2p/src/net/mock/socket.rs:25]: This should be part of constants, not mocks.

**Review by TheQuantumPhysicist — APPROVED** [review]
To be honest, there's not much I can say here until I see this working to replace the current backend of the unit tests.

File: p2p/src/swarm/tests/ban.rs [hunk 2]
```diff
@@ -51,11 +54,26 @@ async fn ban_connected_peer() {
...
 
 #[tokio::test]
-async fn banned_peer_attempts_to_connect() {
+async fn ban_connected_peer_libp2p() {
+    ban_connected_peer::<Libp2pService>(make_libp2p_addr(), make_libp2p_addr()).await;
+}
+
+#[tokio::test]
+async fn ban_connected_peer_mock() {
+    // TODO: implement `ban_peer()`
```
Comment (stanislav-tkach) [inline p2p/src/swarm/tests/ban.rs:63]: Should we create an issue in such cases? In order to not forget about it.

**Review by stanislav-tkach — APPROVED** [review]

File: p2p/src/swarm/tests/ban.rs (hunk 2 above)
Comment (altonen) [reply p2p/src/swarm/tests/ban.rs:63]: good idea, #326 tracks this issue

**Pull request merged**